src/main/java/com/betadevels/onlineshopping/db/hibernate/SubscriptionHibernateDAO.java

```diff
@@ -34,4 +37,24 @@ public Long countOfCustomer(Customer customer)
 		}
 		return count();
 	}
+	public Optional<Subscription> findById(Long id)
```

Comment:
findById is present in the BaseHibernateDAO and hence available to all the models' individual DAOs as they all extend the BaseHibernateDAO. No need to implement it again.